internal/daemon/dpusidemanager.go

```diff
@@ -57,7 +59,7 @@ func (s *DpuSideManager) DeleteBridgePort(context context.Context, bpr *pb.Delet
 	return &emptypb.Empty{}, err
 }
 
-func NewDpuSideManger(vsp plugin.VendorPlugin, config *rest.Config, opts ...func(*DpuSideManager)) *DpuSideManager {
+func NewDpuSideManager(vsp plugin.VendorPlugin, config *rest.Config, opts ...func(*DpuSideManager)) (*DpuSideManager, error) {
```

Comment:
It's a bit weird that the `NewDpuSideManager` function also starts something, a bit unexpected even. That might cause some issues due to started vsp too early (at the time of calling `NewDpuSideManager` function). Can we have a separate start function so that the creation of the struct isn't tied together with the starting of a d.vsp.Start()

Reply:
Ok, I moved it into to Serve routine with stricter order. Let me know if this is better.